pyterrier/apply_base.py

```diff
@@ -190,12 +191,16 @@ def __init__(self,
             batch_size: The number of results per query to process at once. If None, processes in one batch per query.
         """
         self.fn = fn
+        self.verbose = verbose
         self.batch_size = batch_size
 
     def __repr__(self):
         return "pt.apply.by_query()"
 
     def transform_iter(self, inp: pt.model.IterDict) -> pt.model.IterDict:
+        import collections.abc
+        if self.verbose and isinstance(inp, collections.abc.Sized):
```

Comment:
I don't necessarily see a problem showing a progress bar even if it isn't sized -- tqdm will just show the throughput, which is still helpful.

But I'm happy to go either way on this.